http/middleware.go

```diff
@@ -9,12 +9,22 @@ import (
 	"strings"
 	"time"
 
+	"github.com/influxdata/influxdb/kit/tracing"
 	"go.uber.org/zap"
 )
 
 // Middleware constructor.
 type Middleware func(http.Handler) http.Handler
 
+func traceMW(next http.Handler) http.Handler {
```

Comment:
This is awesome - I know you're just moving it, but glad we've abstracted this into a wrapper like this.

Reply:
yay for adding orthogonal functionality without editing specific call sites 🍔 . This was introduced during initial pkger http server